bin/tritond

```diff
@@ -0,0 +1,313 @@
+#!/bin/python
+
+# -*- coding: utf-8 -*-
+"""
+tritond
+~~~~~~~~
+
+Logging daemon for non-blocking triton events.
+
+Adapted from https://github.com/rhettg/BlueOx/blob/master/bin/oxd
+
+"""
+import argparse
+import errno
+import sys
+import logging
+import signal
+import struct
+import os
+import time
+import datetime
+from collections import defaultdict
+
+import zmq
+
+from triton import nonblocking_stream
+from triton.stream import get_stream
+from triton import config
+
+ENV_VAR_TRITON_CONFIG_PATH = 'TRITON_CONFIG_PATH'
+DEFAULT_CONFIG_PATH = '/postmates/etc/triton.yaml'
+
+# Batch writes to Kinesis; collect messages for this long then write
+POLL_LOOP_TIMEOUT_MS = 100
+
+# How often should we poll our file streams. This is also the flush interval.
+FILE_POLL_INTERVAL = 1.0
+
+# How long before we close an open idle file
+FILE_IDLE_TIMEOUT = 60.0
+
+# If we're asked to shutdown before forwarding everything, we'll wait this long
+LINGER_SHUTDOWN_MSECS = 5000
+
+log = logging.getLogger("triton.d")
+
+_triton_config = None
+_streams = dict()
+
+
+def setup_logging(options):
+    if len(options.verbose) > 1:
+        level = logging.DEBUG
+    elif options.verbose:
+        level = logging.INFO
+    else:
+        level = logging.WARNING
+
+    log_format = "%(asctime)s %(levelname)s:%(name)s: %(message)s"
+
+    # We just log straight to stdout. Generally tritond is run by some process
+    # that handles collecting logging for you, like upstart or supervisord. It
+    # would be easy enough to add an option if it was needed by someone though.
+    logging.basicConfig(level=level, format=log_format, stream=sys.stdout)
+
+
+class LogFileStream(object):
```

Comment:
I left it out initially, then put it in solely for testing and debugging purposes (see `class NonblockingStreamEndToEnd(TestCase)` in tests/test_nonblocking_stream.py). Seems a little heavy-weight just for that, but it worked reasonably well. I trimmed down the functionality a bit where obviously un-necessary.

I think an alternative is to clean up the tritond code a bit so it is more feasible to unit test it, removing the need for end-to-end tests.